servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/RoundRobinSelector.java

```diff
@@ -91,6 +91,9 @@ protected Single<C> selectConnection0(
             if (failOpen && failOpenHost == null && host.canMakeNewConnections()) {
                 failOpenHost = host;
             }
+
+            // let the scheduler attempt to skip this host for fairness reasons.
+            scheduler.foundUnhealthy(localCursor);
```

Comment:
Do you think we should update the index only when `host.isHealthy()` and `i > 0`? It may help to reduce access to the global cursor if we update at the end instead of updating on every iteration.

Reply:
I don't think we can skip the first iteration which would be i == 0.

Reply:
Right now you invoke `foundUnhealthy` on every loop iteration. What I propose is to invoke it only if we went inside `return result` branch and at that time `i > 0` (which means we looped at least once). In that case you won't skip anything, but will reduce number of `foundUnhealthy` invocations.

Reply:
I think I got what you're suggesting. I'm not strongly opinionated about it: a few extra atomic int operations are will be a drop in the ocean, but it wasn't significantly much more complex so I did it. lmk if this isn't what you wanted.